src/main/java/backend/resource/Model.java

```diff
@@ -211,6 +211,22 @@ private void ______OPERATIONS_____() {}
                 () -> logger.error("Issue " + issueId + " not found in model for " + repoId));
     }
 
+    /**
+     * Replaces the milestone of an issue specified by {@code issueId} with {@code milestone}
+     * @param issueId
+     * @param milestone
+     * @return the modified TurboIssue if successful
+     */
+    public synchronized Optional<TurboIssue> replaceIssueMilestone(int issueId, Integer milestone) {
+        Optional<TurboIssue> issueLookUpResult = getIssueById(issueId);
+        return Utility.safeFlatMapOptional(issueLookUpResult,
```

Comment:
I know this is copied from existing code, but I'm beginning to think flatmapping here is a bad idea, as the issue-not-found case is a programmer error. Could we take the chance to change this to an exception, and make sure it's handled somewhere? Then everything else on top of this won't have to propagate the optional.